integration-testing/rnode_testing/wait.py

```diff
@@ -64,14 +64,23 @@ def go(): return node.logs()
     return go
 
 
-def show_blocks(node):
+def show_blocks(node, blocks_num):
+    # block_hash re
+    rx = re.compile(r"blockHash: \"([a-zA-Z0-9]*)\"")
+
     def go():
-        exit_code, output = node.show_blocks()
+        exit_code, output = node.show_blocks(blocks_num)
 
         if exit_code != 0:
             raise Exception("Show-blocks failed")
-
-        return output
+        block_infos = ''
+        for block_hash in rx.findall(output):
+            exit_code , block_info = node.show_block(block_hash)
+            if exit_code != 0:
+                raise Exception(f"Show-block {block_hash} failed")
```

Comment:
exit_code and command output for `node.show_block` are relevant here. Please either log them or include them in the exception

Reply:
the `node.show_block` function use `node.exec_run` function which would log the exit_code and the command output. It would be duplicate to log here.

Reply:
sounds good